virttest/libvirt_xml/vm_xml.py

```diff
@@ -3697,20 +3697,20 @@ def marshal_from_iothreads(item, index, libvirtxml):
         """
         Convert a string to iothread tag and attributes.
         """
-        del index
-        del libvirtxml
-        return ('iothread', {'id': item})
+        if not isinstance(item, dict):
+            raise xcepts.LibvirtXMLError("Expected a dictionary of iothread "
+                                         "attributes, not a %s"
+                                         % str(item))
+        return ('iothread', dict(item))
 
     @staticmethod
     def marshal_to_iothreads(tag, attr_dict, index, libvirtxml):
         """
         Convert a iothread tag and attributes to a string.
         """
-        del index
-        del libvirtxml
         if tag != 'iothread':
             return None
-        return attr_dict['id']
+        return dict(attr_dict)
```

Comment:
libvirt/tests/src/virsh_cmd/domain/virsh_iothreadadd.py
there are some python files to depend on VMIothreadidsXML in tp-libvirt,
Could you check whether the change make it compatible with current usage in tp-libvirt ? since in my local attempt, there may not

Reply:
@chunfuwen , see above history. https://github.com/autotest/tp-libvirt/pull/4502 is for other cases' updates to align with this enhancement.